lib/money/money.rb

```diff
@@ -26,10 +26,15 @@ def new(value = 0, currency = nil)
     end
     alias_method :from_amount, :new
 
-    def zero(currency = NULL_CURRENCY)
-      new(0, currency)
+    # A Money of value zero and no 'real' currency. Warning: this may have
+    # surprising behaviour, such as database validations or GraphQL enum not
+    # allowing the string representation of `NullCurrency`. Prefer
+    # `Money.new(0, currency)` to avoid this ambiguity.
+    #
+    # @return [Money]
+    def ambiguous_currencyless_zero
+      new(0, NULL_CURRENCY)
     end
-    alias_method :empty, :zero
```

Comment:
won't this break existing code?

Reply:
We are currently on 0.14.1, so we can still discuss and clean up the uglier parts of the API without problems while adhering to SemVer. There's plenty to do before a 1.0 such as making currency a required argument and disallowing cross-currency computation by default, but I felt this was worth tackling as well.

If easy of upgrading is a concern, I'd be happy to write a Rubocop rule for autocorrecting these.